tests/test_time.py

```diff
@@ -58,6 +58,51 @@ def test_map_years(self):
         )
         assert np.array_equal(years, expected)
 
+    def test_map_year_to_data(self):
+        cal = AdventCalendar(anchor_date=(12, 31), freq="180d")
+        # single year in input data
+        time_index = pd.date_range('20210401', '20211231', freq='180d')
+        test_data = np.random.random(len(time_index))
+        timeseries = pd.Series(test_data, index=time_index)
+        year = cal.map_year_to_data(timeseries)
+
+        expected = np.array(
+            [
+                interval("2021-07-04", "2021-12-31"),
+                interval("2021-01-05", "2021-07-04"),
+            ]
+        )
+
+        assert np.array_equal(year, expected)
+
+        # multiple years in input data
+        time_index = pd.date_range('20200401', '20211231', freq='180d')
+        test_data = np.random.random(len(time_index))
+
+        # timeindex normal order
+        timeseries = pd.Series(test_data, index=time_index)
+        years = cal.map_year_to_data(timeseries)
+        expected = np.array(
+            [
+                [
+                    interval("2021-07-04", "2021-12-31"),
+                    interval("2021-01-05", "2021-07-04"),
+                ],
+                [
+                    interval("2020-07-04", "2020-12-31"),
+                    interval("2020-01-06", "2020-07-04"),  # notice the leap day
+                ],
+            ]
+        )
+
+        assert np.array_equal(years, expected)
+
+        # timeindex reverse order
+        timeseries = pd.Series(test_data, index=time_index[::-1])
+        years = cal.map_year_to_data(timeseries)
+
+        assert np.array_equal(years, expected)
```

Comment:
Add a test case for when the test data ends before the anchor date. E.g. anchor date is 30 Nov, test data ends in August 2020. Then the entire year 2020 should be discarded, but the current implementation will keep it.

Reply:
Thanks for the comment! Now tests concerning edge values are added.